facilitator/src/bin/facilitator.rs

```diff
@@ -926,6 +928,40 @@ fn main() -> Result<(), anyhow::Error> {
     result
 }
 
+/// Check batch signing and packet encryption public keys in this instance's
+/// specific manifests against the corresponding private keys provided. Returns
+/// an error unless each advertised public key matches up with an available
+/// private key.
+fn crypto_self_check(matches: &ArgMatches, logger: &Logger) -> Result<()> {
+    let instance_name = matches.value_of("instance-name").unwrap();
+    let own_manifest = match matches.value_of("own-manifest-base-url") {
+        Some(manifest_base_url) => {
+            SpecificManifest::from_https(manifest_base_url, instance_name, logger)?
+        }
+        // Skip crypto self check if no own manifest is provided
+        None => return Ok(()),
+    };
+
+    let batch_signing_key = batch_signing_key_from_arg(matches)?;
+    own_manifest.verify_batch_signing_key(&batch_signing_key)?;
+    info!(logger, "batch singing key self check OK!");
+
+    let packet_decryption_keys: Vec<PrivateKey> = matches
+        .values_of("packet-decryption-keys")
+        .unwrap()
+        .map(|k| {
+            PrivateKey::from_base64(k)
+                .context("could not parse encoded packet encryption key")
+                .unwrap()
+        })
+        .collect();
+
+    own_manifest.verify_packet_encryption_keys(&packet_decryption_keys)?;
+    info!(logger, "packet decryption key self check OK!");
```

Comment:
these info! logs seem like they're maybe more debug logs? at info level ISTM we'd assume they passed.

Reply:
Good point. These were useful while I was testing this but mostly they're noise. Though it was nice to be logging about something going right for once...